pkg/agent/cniserver/ipam/antrea_ipam_test.go

```diff
@@ -177,10 +391,28 @@ func TestAntreaIPAMDriver(t *testing.T) {
 		require.Error(t, err, "expected error in Add call")
 	}
 
-	testDel := func(test string) {
+	testDel := func(test string, isReserved bool) {
 		owns, err := testDriver.Del(cniArgsMap[test], k8sArgsMap[test], networkConfig)
 		assert.True(t, owns)
 		require.NoError(t, err, "expected no error in Del call")
+
+		podNamespace := string(k8sArgsMap[test].K8S_POD_NAMESPACE)
+		podName := string(k8sArgsMap[test].K8S_POD_NAME)
+		err = wait.Poll(time.Millisecond*200, time.Second, func() (bool, error) {
+			ipPool, _ := antreaIPAMController.ipPoolLister.Get(podNamespace)
+			found := false
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if ipAddress.Owner.Pod != nil && ipAddress.Owner.Pod.Name == podName && ipAddress.Owner.Pod.Namespace == podNamespace {
+					t.Errorf("IP allocation is not removed")
```

Comment:
Isn't the removal async? If it expects the IP to be removed immediately, `wait.Poll` doesn't make sense. If it doesn't expect immediate removal, `t.Errorf` doesn't make sense.

Reply:
It's synced, but we need to wait informer synced in UT.
I've already fixed a similar issue in Traceflow: #3040

Reply:
Then it shouldn't use `t.Errorf` if it's expected to avoid confusion?

Reply:
Thanks. It's wrong code here. Changed to:
```
					t.Logf("IP allocation is not removed")
					return false, nil
```